### ISSUE TITLE
Use DistSQL add resource have unexpected behavior

### ISSUE THREAD

**Issue by wsm12138** [apache/shardingsphere#19224]
## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?

`e6328cb989cfb789189227892cb892740095d2d6`

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
add resource have error .so `show database resources` should't have result
### Actual behavior
add resource have error . excute `show database resources` have result
<img width="1855" alt="image" src="https://user-images.githubusercontent.com/86462784/180726407-73712d41-05f5-480a-a8f0-73de8e5dd525.png">

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
1. use server.yaml start proxy
server.yaml
```
mode:
  type: Cluster
  repository:
    type: ZooKeeper
    props:
      namespace: 0715distsql
      server-lists: 192.168.10.21:2181
      retryIntervalMilliseconds: 500
      timeToLiveSeconds: 60
      maxRetries: 3
      operationTimeoutMilliseconds: 500
  overwrite: true

rules:
- !AUTHORITY
  users:
    - root@%:root
    - sharding@:sharding
  provider:
    type: ALL_PERMITTED
- !SQL_PARSER
  sqlCommentParseEnabled: true
  sqlStatementCache:
    initialCapacity: 2000
    maximumSize: 65535
  parseTreeCache:
    initialCapacity: 128
    maximumSize: 1024

props:
  proxy-frontend-database-protocol-type: "PostgreSQL"
```
2. psql -U root  -h 127.0.0.1 -p 3333(proxy  port) -d postgres
3. execute sql
```
ADD RESOURCE resource_0 (
 URL="jdbc:postgresql://192.168.10.21:5432/postgres",
 USER=gaussdb,
 PASSWORD=passwd
);
```
```
show database resources;
```

**Comment by RaigorJiang** [comment]
@wsm12138 
Sorry, I can't reproduce the exception.
> Use the openGauss database and operate through the postgresql protocol

If you find related problems again, please provide Proxy logs for troubleshooting, thank you!

**Comment by RaigorJiang** [comment]
<img width="628" alt="image" src="https://user-images.githubusercontent.com/5668787/179897433-e0093b46-0982-4f06-af26-52c96c548c7b.png">

**Comment by wsm12138** [comment]
<img width="1681" alt="image" src="https://user-images.githubusercontent.com/86462784/179918206-16624db1-ef2d-46a2-b285-59b784d5327d.png">

### stdout.log

```
[ERROR] 2022-07-20 14:58:22.059 [Connection-2-ThreadExecutor] o.a.s.p.b.t.d.r.r.AddResourceBackendHandler - Add resource failed
java.lang.reflect.InvocationTargetException: null
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.swapper.DataSourceSwapper.setProperties(DataSourceSwapper.java:118)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.swapper.DataSourceSwapper.swap(DataSourceSwapper.java:57)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.XADataSourceFactory.build(XADataSourceFactory.java:43)
        at org.apache.shardingsphere.transaction.xa.jta.datasource.XATransactionDataSource.<init>(XATransactionDataSource.java:65)
        at org.apache.shardingsphere.transaction.xa.XAShardingSphereTransactionManager.newXATransactionDataSource(XAShardingSphereTransactionManager.java:62)
        at org.apache.shardingsphere.transaction.xa.XAShardingSphereTransactionManager.lambda$init$0(XAShardingSphereTransactionManager.java:56)
        at java.util.ArrayList.forEach(ArrayList.java:1259)
        at org.apache.shardingsphere.transaction.xa.XAShardingSphereTransactionManager.init(XAShardingSphereTransactionManager.java:56)
        at org.apache.shardingsphere.transaction.ShardingSphereTransactionManagerEngine.lambda$init$0(ShardingSphereTransactionManagerEngine.java:66)
        at java.util.Map.forEach(Map.java:630)
        at org.apache.shardingsphere.transaction.ShardingSphereTransactionManagerEngine.init(ShardingSphereTransactionManagerEngine.java:66)
        at org.apache.shardingsphere.transaction.rule.TransactionRule.createTransactionManagerEngine(TransactionRule.java:76)
        at org.apache.shardingsphere.transaction.rule.TransactionRule.rebuildEngine(TransactionRule.java:110)
        at org.apache.shardingsphere.transaction.rule.TransactionRule.addResource(TransactionRule.java:87)
        at org.apache.shardingsphere.mode.manager.ContextManager.lambda$updateResources$3(ContextManager.java:208)
        at java.lang.Iterable.forEach(Iterable.java:75)
        at org.apache.shardingsphere.mode.manager.ContextManager.updateResources(ContextManager.java:208)
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:63)
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:44)
        at org.apache.shardingsphere.proxy.backend.text.DatabaseRequiredBackendHandler.execute(DatabaseRequiredBackendHandler.java:51)
        at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.simple.PostgreSQLComQueryExecutor.execute(PostgreSQLComQueryExecutor.java:79)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:108)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: URL invalid jdbc:opengauss://[ip:5432/postgres](http://ip:5432/postgres)
        at org.postgresql.ds.common.BaseDataSource.setUrl(BaseDataSource.java:1313)
        ... 30 common frames omitted
[ERROR] 2022-07-20 14:58:22.060 [Connection-2-ThreadExecutor] o.a.s.p.f.c.CommandExecutorTask - Exception occur:
org.apache.shardingsphere.infra.distsql.exception.resource.InvalidResourcesException: Can not process invalid resources, error messages are: [null].
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:68)
        at org.apache.shardingsphere.proxy.backend.text.distsql.rdl.resource.AddResourceBackendHandler.execute(AddResourceBackendHandler.java:44)
        at org.apache.shardingsphere.proxy.backend.text.DatabaseRequiredBackendHandler.execute(DatabaseRequiredBackendHandler.java:51)
        at org.apache.shardingsphere.proxy.frontend.postgresql.command.query.simple.PostgreSQLComQueryExecutor.execute(PostgreSQLComQueryExecutor.java:79)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.executeCommand(CommandExecutorTask.java:108)
        at org.apache.shardingsphere.proxy.frontend.command.CommandExecutorTask.run(CommandExecutorTask.java:78)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

### server.yaml
```
mode:
  type: Cluster
  repository:
    type: ZooKeeper
    props:
      namespace: 0720distsql
      server-lists: 192.168.10.21:2181
      retryIntervalMilliseconds: 500
      timeToLiveSeconds: 60
      maxRetries: 3
      operationTimeoutMilliseconds: 500
  overwrite: true

rules:
- !AUTHORITY
  users:
    - root@%:root
    - sharding@:sharding
  provider:
    type: ALL_PERMITTED
- !SQL_PARSER
  sqlCommentParseEnabled: true
  sqlStatementCache:
    initialCapacity: 2000
    maximumSize: 65535
  parseTreeCache:
    initialCapacity: 128
    maximumSize: 1024

props:
  proxy-frontend-database-protocol-type: "PostgreSQL"
```

**Comment by wsm12138** [comment]
### About zk(we found content conflicting betweem `DistSQL` & `zk`)
<img width="1084" alt="image" src="https://user-images.githubusercontent.com/86462784/179919562-de1a947e-51e2-4dcb-bf29-dec1e91d6704.png">

**Comment by jingshanglu** [comment]
@wsm12138 When adding a resource, an error will be reported when using `opengauss`, but no error will be reported when using `postgresql`, and the protocol in the url should be consistent with the configuration.

**Comment by jingshanglu** [comment]
Now, there should be a check that prohibits adding resources of different protocols.

**Comment by RaigorJiang** [comment]
Hi @jingshanglu 
Thanks for your investigation, I will assign this issue to you.

**Comment by wsm12138** [comment]
## ShardingSphere version
```
ShardingSphere-5.1.3-SNAPSHOT
Commit ID: 6fa502a13bfe9a4bd2769f742c40677b6f3bef85
Commit Message: Fix #19224 (#19445)
Branch: master
Build time: 2022-07-25T14:34:32+0800
```
## I verify this pr, the problem still exist
<img width="1815" alt="image" src="https://user-images.githubusercontent.com/86462784/180716721-e8074ae1-6f91-4124-b58e-060a5e712e6a.png">
<img width="794" alt="image" src="https://user-images.githubusercontent.com/86462784/180717328-13a30e28-d02b-4446-ad0e-aa2697a9f514.png">

## When we use this version proxy,execute `ADD resource` we can see inconsistency between `proxy` & `zk`

**Comment by wsm12138** [comment]
## server.yaml
```
mode:
  type: Cluster
  repository:
    type: ZooKeeper
    props:
      namespace: 0725distsql
      server-lists: 192.168.10.21:2181
      retryIntervalMilliseconds: 500
      timeToLiveSeconds: 60
      maxRetries: 3
      operationTimeoutMilliseconds: 500
  overwrite: true

rules:
- !AUTHORITY
  users:
    - root@%:root
    - sharding@:sharding
  provider:
    type: ALL_PERMITTED
- !SQL_PARSER
  sqlCommentParseEnabled: true
  sqlStatementCache:
    initialCapacity: 2000
    maximumSize: 65535
  parseTreeCache:
    initialCapacity: 128
    maximumSize: 1024

props:
  proxy-frontend-database-protocol-type: "PostgreSQL"
```
## DistSQL
```
postgres=> create database test;
CREATE DATABASE
postgres=> \c test
psql.bin (10.17, server 12.3-ShardingSphere-Proxy 5.1.3-SNAPSHOT-6fa502a)
WARNING: psql.bin major version 10, server major version 12.
         Some psql features might not work.
You are now connected to database "test" as user "root".
test=> ADD RESOURCE resource_0 (
test(>  URL="jdbc:postgresql://192.168.10.21:5432/postgres",
test(>  USER=gaussdb,
test(>  PASSWORD=sphereEx@2021
test(> );
SUCCESS
test=> show database resources;
    name    |    type    |     host      | port |    db    | connection_timeout_milliseconds | idle_timeout_milliseconds | max_lifetime_milliseconds | max_pool_size | min_pool_size | read_only |
                                                           other_attributes
------------+------------+---------------+------+----------+---------------------------------+---------------------------+---------------------------+---------------+---------------+-----------+-----------------------------------------------------------------
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 resource_0 | PostgreSQL | 192.168.10.21 | 5432 | postgres | 30000                           | 60000                     | 2100000                   | 50            | 1             | false     | {"dataSourceProperties":{},"healthCheckProperties":{},"initializ
ationFailTimeout":1,"validationTimeout":5000,"leakDetectionThreshold":0,"poolName":"HikariPool-2","registerMbeans":false,"allowPoolSuspension":false,"autoCommit":true,"isolateInternalQueries":false}
(1 row)

test=> ADD RESOURCE resource_1 (
test(>  URL="jdbc:opengauss://192.168.10.21:5432/postgres",
test(>  USER=gaussdb,
test(>  PASSWORD=sphereEx@2021
test(> );
ERROR:  java.lang.reflect.InvocationTargetException
test=> show database resources;
    name    |   type    |     host      | port |    db    | connection_timeout_milliseconds | idle_timeout_milliseconds | max_lifetime_milliseconds | max_pool_size | min_pool_size | read_only |
                                                          other_attributes
------------+-----------+---------------+------+----------+---------------------------------+---------------------------+---------------------------+---------------+---------------+-----------+------------------------------------------------------------------
------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 resource_1 | openGauss | 192.168.10.21 | 5432 | postgres | 30000                           | 60000                     | 2100000                   | 50            | 1             | false     | {"dataSourceProperties":{},"healthCheckProperties":{},"initializa
tionFailTimeout":1,"validationTimeout":5000,"leakDetectionThreshold":0,"poolName":"HikariPool-4","registerMbeans":false,"allowPoolSuspension":false,"autoCommit":true,"isolateInternalQueries":false}
(1 row)

test=> preview select 1;
 data_source_name | actual_sql
------------------+------------
 resource_1       | select 1
(1 row)

test=> ADD RESOURCE resource_2 (
test(>  URL="jdbc:opengauss://192.168.10.21:5432/test",
test(>  USER=gaussdb,
test(>  PASSWORD=sphereEx@2021
test(> );
ERROR:  java.lang.reflect.InvocationTargetException
test=> show database resources;
    name    |   type    |     host      | port |  db  | connection_timeout_milliseconds | idle_timeout_milliseconds | max_lifetime_milliseconds | max_pool_size | min_pool_size | read_only |
                                                      other_attributes
------------+-----------+---------------+------+------+---------------------------------+---------------------------+---------------------------+---------------+---------------+-----------+----------------------------------------------------------------------
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 resource_2 | openGauss | 192.168.10.21 | 5432 | test | 30000                           | 60000                     | 2100000                   | 50            | 1             | false     | {"dataSourceProperties":{},"healthCheckProperties":{},"initialization
FailTimeout":1,"validationTimeout":5000,"leakDetectionThreshold":0,"poolName":"HikariPool-6","registerMbeans":false,"allowPoolSuspension":false,"autoCommit":true,"isolateInternalQueries":false}
(1 row)

test=> preview select 1;
 data_source_name | actual_sql
------------------+------------
 resource_2       | select 1
(1 row)
```
